### PR TITLE
Support buster

### PR DESCRIPTION
**Pull request by patfreeman**
* Use the latest alpha release of d-i
* Add ansible config for ocemr_ansible playbook, no cfengine
* Caching of d-i vmlinuz and initrd.gz
* Support testing hardware and firmwares
* 3 disk config options

### CODE REVIEW

**Comment by patfreeman** [comment]
for issue #4

File: README.md [hunk 1]
```diff
@@ -10,6 +10,12 @@ runs on. You just have to press enter once! Be careful.
 
 #### Obtain a USB Image
 
+###### Configure your server
+
+    $ vi ocemr.yml
+
+Note: This is the ansible configuration file. See [ocemr_ansible](https://github.com/patfreeman/ocemr_ansible) for more information.
```
Comment (ph1l) [inline README.md]: I dig the ansible, why not just include it in this repo?

File: README.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ runs on. You just have to press enter once! Be careful.
 
 #### Obtain a USB Image
 
+###### Configure your server
+
+    $ vi ocemr.yml
```
Comment (ph1l) [inline README.md]: doesn't exist:

    + sudo cp -va ocemr.yml /tmp/11298.usb/ansible/
    cp: cannot stat 'ocemr.yml': No such file or directory

File: setup_installer.sh [hunk 3]
```diff
@@ -50,14 +62,14 @@ sudo grub-install --root-directory=/tmp/$$.usb --modules="ext2 part_msdos" ${LOO
...
+sudo cp initrd.gz /tmp/$$.usb/debian-installer-${DEB_REL}-${ARCH}/initrd.gz
 
 sudo cp -r src/preseed /tmp/$$.usb/
 
-sudo mkdir -p /tmp/$$.usb/repo
-sudo cp -v repo/*_all.deb repo/*_${ARCH}.deb /tmp/$$.usb/repo/
-( cd /tmp/$$.usb; dpkg-scanpackages repo/ /dev/null | sudo tee repo/Packages && sudo gzip repo/Packages )
+sudo mkdir -p /tmp/$$.usb/ansible
+sudo cp -va ansible/ /tmp/$$.usb/ansible/ocemr/
+sudo cp -va ocemr.yml /tmp/$$.usb/ansible/
```
Comment (ph1l) [inline setup_installer.sh]: This file doesn't exist in the repo. If we expect the user to copy it from the other repo please document that and check for the it's missing case before we go through most of the setup script ;-)
    + sudo cp -va ocemr.yml /tmp/11298.usb/ansible/
    cp: cannot stat 'ocemr.yml': No such file or directory

File: src/preseed/ocemr_installer.cfg [hunk 4]
```diff
@@ -1 +1,10 @@
-d-i preseed/include string common.cfg i386.cfg
+# Install on an ultra small disk 4gb
+#d-i preseed/include string common.cfg disk-4g.cfg config.cfg
+
+# Install on any normal single hard drive
+#d-i preseed/include string common.cfg disk-auto.cfg config.cfg
+
+# Install on a software raid
+d-i preseed/include string common.cfg disk-raid.cfg config.cfg
```
Comment (ph1l) [inline src/preseed/ocemr_installer.cfg]: screenshot_ocemr-test_2018-08-14_21 23 52

I guess you're making some assumptions about a specific hardware platform now.

File: src/preseed/rc.local.first [hunk 5]
```diff
@@ -1,40 +1,7 @@
...
-/usr/sbin/cf-agent -IKf /var/cfengine/masterfiles/failsafe.cf
-sleep 2
-/usr/sbin/cf-agent -IK
-sleep 2
-/usr/sbin/cf-agent -IK
-sleep 2
-/usr/sbin/cf-agent -IK
+# Run ansible
+cd /root/ansible/
+ansible-playbook ./ocemr.yml
```
Comment (ph1l) [inline src/preseed/rc.local.first]: `root@ocemr-server:~/ansible# ansible-playbook -vvvD ./ocemr.yml `:

```
TASK [ocemr : Install ocemr] ***********************************************************************************
task path: /root/ansible/ocemr/tasks/main.yml:12
fatal: [localhost]: FAILED! => {
    "msg": "The conditional check 'app_server' failed. The error was: error while evaluating conditional (app_server): 'app_server' is undefined\n\nThe error appears to have been in '/root/ansible/ocemr/tasks/main.yml': line 12, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Install ocemr\n  ^ here\n"
}
        to retry, use: --limit @/root/ansible/ocemr.retry

PLAY RECAP *****************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=1   
```

**Comment by ph1l** [comment]
Some other notes, the install seems partial, no apache or ocemr, possibly due to the ansible error? I'm not sure..

File: README.md (hunk 1 above)
Comment (patfreeman) [reply README.md]: Included now.

File: README.md (hunk 2 above)
Comment (patfreeman) [reply README.md]: Included the yml.

File: src/preseed/common.cfg [hunk 6]
```diff
@@ -47,73 +38,7 @@ d-i clock-setup/utc boolean true
...
-
-
-# This command is run just before the install finishes, but when there is
-# still a usable /target directory. You can chroot to /target and use it
-# directly, or use the apt-install and in-target commands to easily install
-# packages and run commands in the target system.
-#d-i preseed/late_command string apt-install zsh; in-target chsh -s /bin/zsh
-
-d-i preseed/late_command string /bin/sh -ex /hd-media/preseed/late_command.sh
+d-i anna/choose_modules string network-console
```
[the remark this answers is not in the document]
Comment (patfreeman) [reply src/preseed/common.cfg]: Removed

File: src/preseed/ocemr_installer.cfg (hunk 4 above)
Comment (patfreeman) [reply src/preseed/ocemr_installer.cfg]: Updated README to add to the configure step.

File: src/preseed/rc.local.first (hunk 5 above)
Comment (patfreeman) [reply src/preseed/rc.local.first]: Should work with the added yml.

**Comment by patfreeman** [comment]
The installer only does the base required to get a running system, then ansible takes over.

**Comment by patfreeman** [comment]
I've got a chicken and egg problem, so I'm going include a minimal playbook to install the appd package, and let it take over from there.

File: setup_installer.sh (hunk 3 above)
Comment (patfreeman) [reply setup_installer.sh]: fixed

File: src/preseed/config.cfg [hunk 7]
```diff
@@ -2,8 +2,8 @@ d-i base-installer/kernel/linux/initramfs-generators string initramfs-tools
 
 # We don't want use a remote mirror (all stuff we need is on the USB flash drive)
 d-i base-installer/install-recommends boolean true
-d-i apt-setup/non-free boolean false
-d-i apt-setup/contrib boolean false
+d-i apt-setup/non-free boolean true
```
Comment (ph1l) [inline src/preseed/config.cfg]: non-free.. ewww... do we really need it? why?

File: src/preseed/config.cfg [hunk 8]
```diff
@@ -15,7 +15,8 @@ d-i debian-installer/allow_unauthenticated boolean true
 tasksel tasksel/first multiselect standard
 d-i pkgsel/include string \
 	bzip2 rsync openssh-server ntp \
-	git ansible gpg
+	git ansible gpg \
+	screen firmware-misc-nonfree intel-microcode
```
Comment (ph1l) [inline src/preseed/config.cfg]: intel-microcode? that's prolly why...

File: src/preseed/rc.local.first [hunk 9]
```diff
@@ -1,11 +1,14 @@
...
+# Wait for a network connection. A minute should do it.
+sleep 60
+
 cd /root/ansible/
 # install appd
 ansible-playbook ./ocemr.yml
+# copy the config
 cp ./ocemr.yml /etc/ocemr-appliance/ansible/
-# install everything
+# install everything else
```
Comment (ph1l) [inline src/preseed/rc.local.first]: :-)

**Comment by ph1l** [comment]
screenshot_ocemr_2018-10-24_16 30 17

neat.. seems to be working!

File: ansible/tasks/main.yml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+---
+# tasks file for ocemr
```
Comment (ph1l) [inline ansible/tasks/main.yml]: Any further thoughts about a shared infrastructure? What about if we use githubs releases feature, to tag and upload debs to github? It wouldn't be apt repos tho... so we'd have to verify packages another way? Or I suppose, we could just setup an s3 (http site) with a repo, and allow multiple people signing rights?

**Comment by ph1l** [comment]
Looking good, but the rc.local sure takes a long time, is it stuck:

```
root       332  0.0  0.0   2372   692 ?        Ss   23:28   0:00 /bin/sh /etc/rc.local start
root      2291 14.8  7.1 182456 72024 ?        Sl   23:30   1:33  \_ /usr/bin/python /usr/bin/ansible-playbook -M /etc/ocemr-appliance/ansible /etc/ocemr-appliance/ansible/ocemr.yml
root      2522  0.0  6.7 182712 67884 ?        S    23:31   0:00      \_ /usr/bin/python /usr/bin/ansible-playbook -M /etc/ocemr-appliance/ansible /etc/ocemr-appliance/ansible/ocemr.yml
root      2537  0.0  0.0   2372   760 ?        S    23:31   0:00          \_ /bin/sh -c /bin/sh -c '/usr/bin/python /root/.ansible/tmp/ansible-tmp-1540423875.98-268710585747251/systemd.py && sleep 0'
root      2538  0.0  0.0   2372   692 ?        S    23:31   0:00              \_ /bin/sh -c /usr/bin/python /root/.ansible/tmp/ansible-tmp-1540423875.98-268710585747251/systemd.py && sleep 0
root      2539  0.0  1.1  22684 11668 ?        S    23:31   0:00                  \_ /usr/bin/python /root/.ansible/tmp/ansible-tmp-1540423875.98-268710585747251/systemd.py
root      2540  0.0  1.4  30464 14940 ?        S    23:31   0:00                      \_ /usr/bin/python /tmp/ansible_a_taq5/ansible_module_systemd.py
root      2542  0.0  0.1  15600  1368 ?        S    23:31   0:00                          \_ /bin/systemctl start serial-getty@ttyS0.service
```

```
ocemr@ocemr-server:~$ sudo systemctl status serial-getty@ttyS0.service 
● serial-getty@ttyS0.service - Serial Getty on ttyS0
   Loaded: loaded (/lib/systemd/system/serial-getty@.service; enabled; vendor preset: enabled)
   Active: inactive (dead)
     Docs: man:agetty(8)
           man:systemd-getty-generator(8)
           http://0pointer.de/blog/projects/serial-console.html
```

**Comment by ph1l** [comment]
> I've got a chicken and egg problem, so I'm going include a minimal playbook to install the appd package, and let it take over from there.

hmm, there are a couple of other approaches you could take:
 - include the `.deb`s needed to install the appd package in the installer image and use late_command to install them in-target

File: src/preseed/config.cfg (hunk 7 above)
Comment (patfreeman) [reply src/preseed/config.cfg]: intel-microcode

File: src/preseed/config.cfg (hunk 8 above)
Comment (patfreeman) [reply src/preseed/config.cfg]: Yup

**Comment by patfreeman** [comment]
> > I've got a chicken and egg problem, so I'm going include a minimal playbook to install the appd package, and let it take over from there.
> 
> hmm, there's another option: include the `.deb`s needed to install the initial appd package in the installer image and use late_command to install them in-target.

I thought about that, but wanted to remove the whole debs in debs stuff. I felt the external repo to pull the latest packages was best.

File: ansible/tasks/main.yml (hunk 10 above)
Comment (patfreeman) [reply ansible/tasks/main.yml]: Yup. a shared S3 site would probably be best (and cheapest). How should we set something up?

**Pull request merged**